enumerable_method.rb

```diff
@@ -0,0 +1,124 @@
+public def my_each(arr)
+    for name in arr do
+        yield (name)
+    end
+end
+
+names = ["Amy", "John", "Joe"]
+my_each(names) {|a| puts "Hello #{a}"}
+
+
+names = %w[Amy John Joe]
+
+my_each(names) { |a| puts "Hello #{a}" }
+
+def my_each_with_index(arr)
+  len = arr.length
+
+  i = 1
+
+  my_each(arr) do |item|
+    yield item if i <= len && i.even?
+
+    i += 1
+  end
+end
```

Comment:
#my_each_with_index returns an enumerator if no block is given
#my_each_with_index does not mutate the original array
#my_each_with_index when a block is given when self is an array returns the array itself after 
calling the given block once for each element in self
#my_each_with_index when a block is given when self is an array behaves like #each_with_index calls the given block once for each element in self
#my_each_with_index when a block is given when self is a range returns the range itself after 
calling the given block once for each element in self
#my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
#my_each_with_index when a block is given when self is a hash behaves like each_with_index calls the given block once for each element in self